src/libmongoc/src/mongoc/mongoc-topology.c

```diff
@@ -1682,3 +1659,165 @@ _mongoc_topology_bypass_cooldown (mongoc_topology_t *topology)
    BSON_ASSERT (topology->single_threaded);
    topology->scanner->bypass_cooldown = true;
 }
+
+static void
+_find_topology_version (const bson_t *reply, bson_t *topology_version)
+{
+   bson_iter_t iter;
+   const uint8_t *bytes;
+   uint32_t len;
+
+   if (!bson_iter_init_find (&iter, reply, "topologyVersion") ||
+       !BSON_ITER_HOLDS_DOCUMENT (&iter)) {
+      bson_init (topology_version);
+      return;
+   }
+   bson_iter_document (&iter, &len, &bytes);
+   bson_init_static (topology_version, bytes, len);
+}
+
+
+/* "Clears" the connection pool by incrementing the generation.
+ * 
+ * Pooled clients with open connections will discover the invalidation
+ * the next time they fetch a stream to the server.
+ * 
+ * Caller must lock topology->mutex. */
+void
+_mongoc_topology_clear_connection_pool (mongoc_topology_t *topology,
+                                        uint32_t server_id)
+{
+   mongoc_server_description_t *sd;
+   bson_error_t error;
+
+   sd = mongoc_topology_description_server_by_id (
+      &topology->description, server_id, &error);
+   if (!sd) {
+      /* Server removed, ignore and ignore error. */
+      return;
+   }
+   TRACE ("clearing pool for server: %s", sd->host.host_and_port);
+   sd->generation++;
```

Comment:
I may be missing it, but where is the cluster's `generation` field ever incremented?